synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -225,6 +226,34 @@ async def action_for_event_by_user(
             sender_power_level,
         ) = await self._get_power_levels_and_sender_level(event, context)
 
+        related_events: Dict[str, Dict[str, str]] = {}
+        if self._related_event_match_enabled:
+            related_event_id = event.content.get("m.relates_to", {}).get("event_id")
+            relation_type = event.content.get("m.relates_to", {}).get("rel_type")
+            if related_event_id is not None and relation_type is not None:
+                related_event = await self.store.get_event(
+                    related_event_id, allow_none=True
+                )
+                if related_event is not None:
+                    related_events[relation_type] = _flatten_dict(related_event)
+
+            reply_event_id = (
+                event.content.get("m.relates_to", {})
+                .get("m.in_reply_to", {})
+                .get("event_id")
+            )
+            if reply_event_id is not None and (
+                relation_type != "m.thread"
+                or not event.content.get("m.relates_to", {}).get(
+                    "is_falling_back", False
+                )
+            ):
+                related_event = await self.store.get_event(
+                    reply_event_id, allow_none=True
+                )
+                if related_event is not None:
+                    related_events["m.in_reply_to"] = _flatten_dict(related_event)
```

Comment:
I think its probably worth pulling this out into a helper function? It's a bit dense to be in the middle of this already fairly long function

Reply:
Sure, sounds good to me!